### PR TITLE
Add ability to set a custom subnet when creating a nodegroup

### PR DESCRIPTION
**Pull request by Skarlso**
### Description

Closes https://github.com/weaveworks/eksctl/issues/3824.

### TODO

- [ ] Update integration tests 
- [ ] Update docs

### Checklist
- [x] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes
- [x] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: pkg/ctl/create/nodegroup.go [hunk 1]
```diff
@@ -30,6 +31,11 @@ func createNodeGroupCmd(cmd *cmdutils.Cmd) {
 			return api.ErrInvalidName(ng.Name)
 		}
 
+		if options.SubnetID != "" {
+			// TODO: validate the subnet being part of the VPC?
```
Comment (aclevername) [inline pkg/ctl/create/nodegroup.go:35]: mhmm we could do that, or let the natural CFN error bubble up.

Comment (Skarlso) [reply pkg/ctl/create/nodegroup.go:35]: Oh, I actually do this already in another part of the validation. I guess I don't need this TODO any longer.

Comment (Skarlso) [reply pkg/ctl/create/nodegroup.go:35]: https://github.com/weaveworks/eksctl/pull/4025/files#diff-84e0553c6ff9569d22f56a2a8038897508f76b733dd2304e73ee29f9c9ca1e69R608 herre

File: integration/tests/crud/creategetdelete_test.go [hunk 2]
```diff
@@ -274,6 +276,110 @@ var _ = Describe("(Integration) Create, Get, Scale & Delete", func() {
...
+					AvailabilityZone: aws.String("us-west-2a"),
+					CidrBlock:        aws.String(cidr),
+					TagSpecifications: []*awsec2.TagSpecification{
+						{
+							ResourceType: aws.String(awsec2.ResourceTypeSubnet),
+							Tags:         tags,
+						},
+					},
+					VpcId: s.VpcId,
+				})
```
Comment (aclevername) [inline integration/tests/crud/creategetdelete_test.go:342]: does this need its own unique cleanup for the `DeleteCluster` to work? or is it alright?

Comment (Skarlso) [reply integration/tests/crud/creategetdelete_test.go:342]: That's a good question. I didn't try that out. I'll try it out.

**Review by aclevername — APPROVED** [review]

File: integration/tests/crud/creategetdelete_test.go (hunk 2 above)
Comment (Skarlso) [reply integration/tests/crud/creategetdelete_test.go:342]: worked! :) 

Added the subnet:

```
2021-08-02 15:21:06 [ℹ]  node "ip-192-169-16-174.us-west-2.compute.internal" is ready
2021-08-02 15:21:06 [ℹ]  node "ip-192-169-254-82.us-west-2.compute.internal" is ready
2021-08-02 15:21:06 [✔]  created 1 managed nodegroup(s) in cluster "test-adding-subnets-9"
2021-08-02 15:21:08 [ℹ]  checking security group configuration for all nodegroups
2021-08-02 15:21:08 [ℹ]  all nodegroups have up-to-date configuration
```

Deleted the cluster:

```
2021-08-02 15:30:33 [ℹ]  waiting for CloudFormation stack "eksctl-test-adding-subnets-9-nodegroup-managed-ng-1"
2021-08-02 15:30:51 [ℹ]  waiting for CloudFormation stack "eksctl-test-adding-subnets-9-nodegroup-managed-ng-1"
2021-08-02 15:31:09 [ℹ]  waiting for CloudFormation stack "eksctl-test-adding-subnets-9-nodegroup-managed-ng-1"
2021-08-02 15:31:10 [ℹ]  will delete stack "eksctl-test-adding-subnets-9-cluster"
2021-08-02 15:31:11 [✔]  all cluster resources were deleted

```

**Comment by Callisto13** [comment]
docs coming in a separate pr?

**Comment by Skarlso** [comment]
@Callisto13 err..... yes. Definitely!

**Pull request merged**